lib/puppet/provider/cisco_pim/nxapi.rb

```diff
@@ -0,0 +1,120 @@
+#
+# The NXAPI provider for cisco_pim.
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+Puppet::Type.type(:cisco_pim).provide(:nxapi) do
+  desc 'The NXAPI provider for cisco_pim.'
+
+  confine feature: :cisco_node_utils
+
+  mk_resource_methods
+
+  PIM_ALL_PROPS = [
```

Comment:
This provider will eventually get a lot more properties. Can you code it like the other providers? e.g.:

```
PIM_NON_BOOL_PROPS = [
  :ssm_range
]
PIM_BOOL_PROPS = [
]
PIM_ALL_PROPS = PIM_NON_BOOL_PROPS + PIM_BOOL_PROPS

# Dynamic method generation for getters & setters
PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@pim',
                                            PIM_NON_BOOL_PROPS)
PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@pim',
                                            PIM_BOOL_PROPS)
```

The bool props will just be noops for now.